src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -1736,7 +1668,7 @@ grpc_error_handle ClientChannel::DoPingLocked(grpc_transport_op* op) {
           ABSL_EXCLUSIVE_LOCKS_REQUIRED(&ClientChannel::work_serializer_) {
             SubchannelWrapper* subchannel = static_cast<SubchannelWrapper*>(
                 complete_pick->subchannel.get());
-            ConnectedSubchannel* connected_subchannel =
+            RefCountedPtr<ConnectedSubchannel> connected_subchannel =
```

Comment:
This function is called `DoPingLocked` - I assume that it's a different lock and that there's a strict ordering that is being observed here?